projects/RabbitMQ.Client/client/impl/RecoveryAwareChannel.cs

```diff
@@ -79,6 +98,23 @@ public override void BasicNack(ulong deliveryTag, bool multiple, bool requeue)
             }
         }
 
+        public override ValueTask BasicNackAsync(ulong deliveryTag, bool multiple, bool requeue)
+        {
+            ulong realTag = deliveryTag - ActiveDeliveryTagOffset;
+            if (realTag > 0 && realTag <= deliveryTag)
+            {
+                return base.BasicNackAsync(realTag, multiple, requeue);
+            }
+            else
+            {
+#if NET6_0_OR_GREATER
```

Comment:
Technically these are not the same, ValueTask.CompletedTask and ValueTask(Task.CompletedTask).
I'd just do a return default in all cases and not do the #if. That should work fine for .netstandard too.

https://github.com/dotnet/runtime/blob/0151c4696eb14f52f567ceee91d2896ff2586edf/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ValueTask.cs#L113

Or do a polyfill style helper somewhere if you prefer that style.

Reply:
Thank you for noticing, and please feel free to review the rest of this PR.